qemu/tests/cfg/usb.cfg

```diff
@@ -214,7 +214,7 @@
                     usb_repeat_times = 1
                 - usb_multi_times:
                     no usb_negative_test
-                    usb_repeat_times = 5000
+                    usb_repeat_times = 20
```

Comment:
20 is too small in my opinion, 20 times is kind of normal is some case, better over 100 times.

Reply:
Are you really plugging in and out USB device 100x into the same virtual machine in real life?

I mean, I am not going to argue because of this, and I am OK with some compromise (or I can change configs on my own anyway), but this looks like really rare use case. I thought even 20 being too much, to be honest.

Just out of curiosity, do you guys have some statistics about errors caught after 100x versus 20x (or even less)?

Reply:
You change it from 5000 to 20, I reckon it is because as you said 

> Repeat the test for 5000 times takes too long and does not make
sense

I understand it takes long time compare to other cases, but it not too long when you compare the time it would cost by running it manually, but 5000 and 20 dose not at same quantitative level, why it does not make sense?

The problem is you need make sure the it tests enough times to discovery any problems, 20 is not enough for sure, it is normal even for a VM.

Using automation running 5000 times is efficient enough, the purpose of this case is not save time, it is find potential errors. And due to this reason, I think you should keep it 5000 times.

Reply:
I have to respectfully disagree.

Purpose of the automation is to speed up the process of testing. To emulate human so that you can save money. Therefore you should emulate the use cases from the real life and put some extra on top of that as stress test. So 5000 is IMHO an overkill.

Also, testing it 5000x when it is killed after 4 hours of running anyway without chance to pass is nonsense still.

Also, please understand that in real life resources are limited. Therefore bug like "mounting USB device fails after 4000x attempt" will receive lower priority. If you report too many such bugs you are actually worsening the situation as it might make somebody overlook more serious bug and load people who have to go through all such bugs.

If you measure the success by number of the reported bugs, your approach is sound. If you want to improve quality of the product, it is not.

But feel free to put there whatever value you desire. I can change it for our setup anyway.